src/Models/ProductAttribute.php

```diff
@@ -0,0 +1,172 @@
+            ])
+            ->groupBy([
+                'eav_attribute.attribute_id',
+                'eav_attribute.attribute_code',
+                'eav_attribute.frontend_input',
+                'eav_attribute.frontend_label',
+                'eav_attribute_label.value'
+            ]);
+        });
+    }
+
+   public function options(): Attribute
+   {
+       return Attribute::make(
+           get: fn () => !$this->option_ids || !$this->option_values ? [] : 
+               (count($optionIds = array_filter(explode(',', $this->option_ids))) === count($optionValues = array_filter(explode(',', $this->option_values))) ? array_combine($optionIds, $optionValues) : [])
+       );
+   }
+
+   public function formattedOptions(): Attribute
+   {
+       return Attribute::make(
+           get: fn () => empty($this->options) ? '' : 
+               collect($this->options)
+                   ->map(fn($value, $id) => "{$value} (ID: {$id})")
+                   ->join(', ')
+       );
+   }
+
+   public function frontendLabel(): Attribute
+   {
+       return Attribute::make(
+           get: fn () => $this->store_frontend_label ?? $this->attributes['frontend_label'] ?? ''
+       );
+   }
+
+    public function getCacheKey(): string
+    {
+        return "product_attribute_{$this->attribute_id}_store_" . static::getCurrentStoreId();
+    }
+
+    public function scopeVisible($query)
+    {
+        return $query->where('is_visible', 1);
+    }
+
+    public function scopeVisibleOnFront($query)
+    {
+        return $query->where('is_visible_on_front', 1);
+    }
+
+    public function scopeFilterable($query)
+    {
+        return $query->where('is_filterable', 1);
+    }
+
+    public function scopeFilterableInSearch($query)
+    {
+        return $query->where('is_filterable_in_search', 1);
+    }
+
+    public function scopeSearchable($query)
+    {
+        return $query->where('is_searchable', 1);
+    }
+
+    public function scopeUsedInProductListing($query)
+    {
+        return $query->where('used_in_product_listing', 1);
+    }
+
+    public function scopeWithOptions($query)
+    {
+        return $query->whereIn('frontend_input', ['select', 'multiselect', 'swatch_visual', 'swatch_text']);
+    }
+
+    public function scopeByInputType($query, $type)
+    {
+        if (!isset(static::$inputTypes[$type])) {
+            return $query;
+        }
+
+        return $query->where('frontend_input', $type);
+    }
+
+    public function attributeOptions()
+    {
+        return $this->hasMany(ProductAttributeOption::class, 'attribute_id', 'attribute_id');
+    }
+
+    protected static function boot()
+    {
+        parent::boot();
+
+        static::saved(function ($model) {
+            Cache::forget($model->getCacheKey());
+        });
+
+        static::deleted(function ($model) {
+            Cache::forget($model->getCacheKey());
+        });
```

Comment:
We could put this into an observer to clean this file up a bit, also you don't have to overwrite the boot method https://laravel.com/docs/11.x/eloquent#observers